pulsar-broker/src/test/java/org/apache/pulsar/broker/service/BrokerEntryMetadataE2ETest.java

```diff
@@ -273,6 +274,32 @@ public void testGetLastMessageId() throws Exception {
                 .subscriptionType(SubscriptionType.Exclusive)
                 .subscriptionName(subscription)
                 .subscribe();
-        consumer.getLastMessageId();
+    }
+
+    @Test(timeOut = 20000)
+    public void testConsumerGetBrokerEntryMetadata() throws Exception {
+        final String topic = newTopicName();
+        final String subscription = "my-sub";
+
+        @Cleanup
+        Producer<byte[]> producer = pulsarClient.newProducer()
+                .topic(topic)
+                .create();
+
+        @Cleanup
+        Consumer<byte[]> consumer = pulsarClient.newConsumer()
+                .topic(topic)
+                .subscriptionType(SubscriptionType.Exclusive)
+                .subscriptionName(subscription)
+                .subscribe();
+
+        long sendTime = System.currentTimeMillis();
+        producer.send("hello".getBytes());
+
+        Message<byte[]> received = consumer.receive();
+        Assert.assertEquals("hello", new String(received.getData()));
+
+        Assert.assertTrue(received.getBrokerPublishTime() >= sendTime);
+        Assert.assertEquals(received.getIndex(), 0);
```

Comment:
Zero is an unfortunate value, can we test some value that we are sure it is not a typical default value?

Reply:
Fixed. Improve it by sending 10 messages and the index should be from 0 to 9.